src/libstd/collections/hash/map.rs

```diff
@@ -716,6 +716,251 @@ impl<K: Hash + Eq, V> HashMap<K, V, RandomState> {
     }
 }
 
+impl<K, V, S> HashMap<K, V, S> {
+    /// Returns a reference to the map's [`BuildHasher`].
+    ///
+    /// [`BuildHasher`]: ../../std/hash/trait.BuildHasher.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::HashMap;
+    /// use std::collections::hash_map::RandomState;
+    ///
+    /// let hasher = RandomState::new();
+    /// let map: HashMap<i32, i32> = HashMap::with_hasher(hasher);
+    /// let hasher: &RandomState = map.hasher();
+    /// ```
+    #[stable(feature = "hashmap_public_hasher", since = "1.9.0")]
+    pub fn hasher(&self) -> &S {
```

Comment:
I would prefer if this function kept the `S: BuildHasher` bound. I have some changes to hashbrown (which will soon replace the libstd hashmap) which may require this bound in the future.

Do you really need to call `map.hasher()` without a `S: BuildHasher` bound? If you really need it then I can try looking into a workaround, but it might get a bit ugly.

Reply:
> Do you really need to call `map.hasher()` without a `S: BuildHasher` bound? If you really need it then I can try looking into a workaround, but it might get a bit ugly.

Nope, it was just one of the methods I saw that didn't need a bound, I can add it back.

Reply:
Please do.

Reply:
I moved it back where it was before with the same bounds.